### DISCUSSION TITLE
Gate

### DISCUSSION THREAD

**Discussion by danilkorotkov** [HomeSpan/HomeSpan#33] [Q&A] [answered]
Hi. Is it possible to add target position characteristic to garage class like covers?

**Comment by HomeSpan** [comment, marked as answer]
Unfortunately, the Characteristics supported by a Service are defined by HomeKit and cannot be changed, unless you create a custom service for use with a custom app (which means it won't work with the Home App or Siri).   For some reason, probably historical, Apple HAP uses TargetDoorPosition for Garage Doors but TargetPosition for Window Coverings.  The latter is an integer ranging from 0-100%, whereas the former (TargetDoorPosition) only supports "Open" and "Closed".

**Reply by danilkorotkov** [reply]
Thank you!

**Reply by danilkorotkov** [reply]
Can I make primary Open-Close method as Hidden on card to control gate though Door states only?

![IMG_853B2F7ED196-1](https://user-images.githubusercontent.com/25537053/107749995-6ecf4c80-6d2c-11eb-85fc-1f2e32dcd9e5.jpeg)

**Comment by HomeSpan** [comment]
Though HomeSpan supports setting a Service to Hidden, I don't think this has any affect on Apple-supported HAP Services.  I'll have to test this further, but I think it's only there to hide non-Apple HAP Services from the Home App that are designed to be used by a separate proprietary app.

**Reply by danilkorotkov** [reply]
Thank you
How to set service as hidden with HAP api? Any example?

**Comment by HomeSpan** [comment]
Use the `setHidden()` method of SpanService.  See https://github.com/HomeSpan/HomeSpan/blob/master/docs/Reference.md#spanservice
